### DISCUSSION TITLE
Problems when using hybrid graphics in Pop OS

### DISCUSSION THREAD

**Discussion by FluxZA** [wezterm/wezterm#2502] [Q&A]
Hi

I have a laptop with a dedicated GPU (RTX 3070). I dual boot Win 10 & Pop-OS. When the graphics mode is set to hybrid in Pop, wezterm is extremely laggy - 1 to 2 seconds lag between input and output on the screen. When using only integrated or dedicated graphics (the laptop has a mux switch) in Pop or any of the 3 options in Windows, there is no lag. I also tried starting wezterm using the 'use dedicated graphics' command from Gnome, but it did not help.

This is not a major problem, as I can use my laptop in either the integrated or dedicated state in most situations. However, I wanted to check that I wasn't missing anything settings wise that could solve my problem.

Thanks in advance for any help.

Some misc info:
Laptop: Asus Tuf A15 2022 model
CPU: AMD Ryzen 7 6800H with Radeon Graphics
Pop version: 22.04 on kernel 5.17.15

**Comment by wez** [comment]
Can you share the opengl version information reporting in the debug overlay in those different scenarios?

Press `CTRl-SHIFT-L` to open the debug overlay and copy and paste the version info here (both the wezterm and opengl info from each scenario).

**Reply by FluxZA** [reply]
Thanks for the quick reply. While gathering the info you asked for I found out that the problem only occurs if Windows is shutdown in hybrid mode. It seems to mess up how Pop discovers and then uses the combined graphics card setup. You can see in the info below that the Open GL version is different based on how Windows is shut down.

Sorry to have brought up what is effectively a Pop/Win issue. Thanks for the awesome terminal.

PS. The info you requested seen as I have it:
wezterm Version for all:
20220905-102802-7d4b8249

Win OpenGL for all combinations:
NVIDIA GeForce RTX 3070 Laptop GPU/PCIe/SSE2 4.5.0 NVIDIA 516.59

Pop OpenGL in Nvidia only mode & Hybrid if Win is shutdown in Nvidia mode:
NVIDIA GeForce RTX 3070 Laptop GPU/PCIe/SSE2 4.6.0 NVIDIA 515.48.07

Pop OpenGL in Integrated & Hybrid if Win is shutdown in Hybrid:
AMD YELLOW_CARP (DRM 3.44.0, 5.17.15-76051715-generic, LLVM 12.0.1) 4.6 (Compatibility Profile) Mesa 21.3.9 (git-78c96ae5b6)
